src/effects.cpp

```diff
@@ -8,6 +8,39 @@ Effects::Effects() {}
 
 Effects::~Effects() {}
 
+bool Effects::effectExists(effectNS::EFFECT_TYPE effectType) {
+	for (effectDurationPair curr : currentEffects) {
+		if (curr.first == effectType)
+			return true;
+	}
```

Comment:
If a non-existence key is accessed,  the map will create a default constructed element for it. Since we are using the `currentEffects` for other forms of checking, we can't have default key value pairs in the map.